File: 1.1/src/cz/mg/vulkan/PFNvkGetDeviceQueue2.java
package cz.mg.vulkan;

public class PFNvkGetDeviceQueue2 extends VkFunctionPointer {
    public PFNvkGetDeviceQueue2() {
    }

    protected PFNvkGetDeviceQueue2(VkMemory vkmemory) {
        super(vkmemory);
    }

    protected PFNvkGetDeviceQueue2(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkGetDeviceQueue2(long value) {
        setValue(value);
    }

    public PFNvkGetDeviceQueue2(VkInstance instance) {
        super(instance, new VkString("vkGetDeviceQueue2"));
    }

    public void call(VkDevice device, VkDeviceQueueInfo2 pQueueInfo, VkQueue pQueue){
        callNative(getValue(), device != null ? device.getVkAddress() : VkPointer.getNullAddressNative(), pQueueInfo != null ? pQueueInfo.getVkAddress() : VkPointer.NULL, pQueue != null ? pQueue.getVkAddress() : VkPointer.NULL);
    }

    protected static native void callNative(long vkaddress, long device, long pQueueInfo, long pQueue);

}

File: 1.1/src/c/cz/mg/vulkan/PFN_vkGetDeviceQueue2.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

void Java_cz_mg_vulkan_PFNvkGetDeviceQueue2_callNative(JNIEnv* env, jclass jc, jlong address, jlong device, jlong pQueueInfo, jlong pQueue){
    (void)env;
    (void)jc;
    PFN_vkGetDeviceQueue2 f = (PFN_vkGetDeviceQueue2)jniLongToFunctionPointer(address);
    f(
        *((VkDevice*)jniLongToPointer(device)),
        ((VkDeviceQueueInfo2*)jniLongToPointer(pQueueInfo)),
        ((VkQueue*)jniLongToPointer(pQueue))
    );
}

